packages/core/parcel-bundler/src/transforms/babel/jsx.js

```diff
@@ -12,6 +12,26 @@ const JSX_PRAGMA = {
   hyperapp: 'h'
 };
 
+function createJSXRegexFor(dependency) {
+  // result looks like /from\s+[`"']react[`"']|require\([`"']react[`"']\)/
+  return new RegExp(
+    `from\\s+[\`"']${dependency}[\`"']|require\\([\`"']${dependency}[\`"']\\)`
+  );
+}
+
+/**
+ * Solves a use case when JSX is used in .js files, but
+ * package.json is empty or missing yet and therefore pragma cannot
+ * be determined based on pkg.dependencies / pkg.devDependencies
+ */
+function maybeCreateFallbackPragma(asset) {
+  for (const dep in JSX_PRAGMA) {
+    if (asset.contents.match(createJSXRegexFor(dep))) {
```

Comment:
Maybe we should create these regexes ahead of time rather than recreating them for every file

Reply:
+1 thanks for the hint, will fix tomorrow morning